libtsuba/include/tsuba/TxnContext.h

```diff
@@ -9,47 +9,35 @@ class KATANA_EXPORT TxnContext {
     node_properties_read_.insert(name);
   }
 
-  void InsertNodePropertyRead(std::unordered_set<std::string> names) {
-    node_properties_read_.insert(names.begin(), names.end());
-  }
-
-  void InsertNodePropertyRead(std::vector<std::string> names) {
+  template <template <class> class Container>
+  void InsertNodePropertyRead(Container<std::string>& names) {
```

Comment:
Let's just take a Range or Iterator:

```
template <typename It>
void InsertNodePropertyRead(It begin, It end) { .... }

template <typename Range>
void InsertNodePropertyRead(const Range& range) {
  InsertNodePropertyRead(range.begin(), range.end());
}
```

and avoid over-parameterizing on things we don't actually know the shape of or even want to control.

Reply:
Sure. Changed using range (but named it as `Container` to be consistent with Emmett's previous PR).